src/users/controllers/MeController.js

```diff
@@ -0,0 +1,18 @@
+import { findUserById } from "../repository/findUserByIdRepository.js";
+
+export class MeController {
+  async handle(req, res) {
+    const user = await findUserById(req.userId);
+
+    if (!user) {
+      return res.status(404).json({
+        error: true,
+        message: "Usuário não encontrado",
+      });
+    }
+
+    return res.status(200).json({
```

Comment:
Não faz sentido colocar o objeto usuário dentro de uma propriedade “usuário”.
A rota `/users/:id` já indica que vc está retornando um elemento da coleção/set de usuário

Reply:
Esqueci de mencionar que, nesse caso, o design das rotas segue um padrão de “sets” e “elementos”

/users (set)
/users/:id (elemento)
/users/:id/projects (set)
/projects (set)
/projects/:id (elemento)

nesse caso, o tipagem do elemento é implícita, garantida apenas por convenção.

Vc também pode customizar o header content-type para conter informações que possam se utilizadas programaticamente a fim de adquirir a tipagem do elemento, mas acredito se desnecessário para a maioria dos projetos

Reply:
> Não faz sentido colocar o objeto usuário dentro de uma propriedade “user”. A rota `/users/:id` já indica que vc está retornando um elemento da coleção/set de usuários

Não entendi isso de colocar o objeto usuário na propriedade user, como eu poderia fazer o retorno ?

Reply:
Acho que entendi o que tu comentou, me corrija se eu estiver errado

Acredito que sua sugestão seja escrever o código dessa forma: 
```js
    return res.status(200).json({
      id: user.id,
      name: user.name,
      email: user.email,
    });
```

para o payload ser desta forma: 
```json
{
    "id": "6112f88e-1c83-474a-bc79-f8146516db5f",
    "name": "bea",
    "email": "bea@gmail.com"
}
```

Pois como sabe que é uma instância de `user` o front-end não vai precisar acessar usando `user.id`, apenas buscar o campo diretamente, né ?

Reply:
@NicolasPereira isso mesmo

`/users` é uma coleção de usuários e que quando vc chama `GET /users/:id` vc está tentando pegar um elemento dessa coleção. Então as duas únicas possibilidades de resultados são "elemento não existe na coleção" (404 - not found) ou "retornar o solicitado elemento da coleção" (200 - ok). Já é esperado que todos os elementos dessa coleção sejam do tipo usuário.